doc/SONiC_long_reset_button_press_hld.md

```diff
@@ -0,0 +1,282 @@
+            feature_state = feature_info.get('feature', {}).get('state')
+            return True if feature_state == 'enabled' else False
+        except Exception:
+            syslog.syslog(syslog.LOG_ERR, "LongRebootPress: Failed to connect to DB")
+            raise
+
+    def start(self):
+        feature_enabled = self.get_feature_state()
+        long_reboot_button_press = get_long_reboot_status()
+
+        if long_reboot_button_press and feature_enabled:
+            syslog.syslog(syslog.LOG_INFO, "LongRebootPress: Long reboot detected, the system is resetting passwords to default")
+            self.platform_long_reboot_press.start()
+        else:
+            syslog.syslog(syslog.LOG_INFO, "LongRebootPress: Skipping..")
+
+
+def main():
+    LongRebootPress().start()
+```
+
+The feature is added to the init flow as long as we set the new ```ENABLE_LONG_RESET_BUTTON_PRESS``` global variable and the exsiting ```CHANGE_DEFAULT_PASSWORD``` global variable in ```rules/config``` to ```y```.
+
+```
+# ENABLE_LONG_RESET_BUTTON_PRESS - enable long reset button press on switch
+ENABLE_LONG_RESET_BUTTON_PRESS ?= y
+```
+
+###
+
+# Performance
+
+The ```long-reboot.service``` is required to run as fast as possible, and it is expected to finish running in milliseconds **(150-300 milliseconds)** on long reset detected, and not delay the init flow more than that.
+
+# Flows
+
+###### Long Reboot Press during init flow
+![](./sonic-reboot-press-init-flow.png  "Long Reboot Press during init")
+
+The defualt long reset button press functionality is to delete non default users and restore original passwords for default users and expire them, all done using linux commands
+
+# Configuration and management
+
+## CLI
+
+<!-- omit in toc -->
+## Command structure
+
+**User interface**:
+```
+config
+\-- long-reset-button-press
+    |-- state <enabled|disabled>
+
+show
+\-- long-reset-button-press
+```
+
+### Config command group
+**The following command set long-reset-button-press  feature state:**
+```bash
+config long-reset-button-press state <enabled|disabled>
+```
+
+### Show command
+**The following command display long-reset-button-press  configuration:**
+```bash
+root@sonic:/home/admin# show long-reset-button-press
+state
+-------
+enabled
+```
+
+## YANG model
+New YANG model `long-reset-button-press.yang` will be added to provide support for configuring long-reset-button-press state.
+
+**Skeleton code:**
+```
+ module sonic-long-reset-button-press {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/sonic-net/long-reset-button-press";
+
+    description "LONG_RESET_BUTTON YANG Module for SONiC-based OS";
+
+    revision 2024-01-04 {
+        description "First Revision";
+    }
+
+    container sonic-long-reset-button-press {
+
+        container LONG_RESET_BUTTON {
+
+            description "LONG_RESET_BUTTON part of config_db.json";
+
+            container STATE {
```

Comment:
Suggest to avoid STATE name for the CONFIG_DB data as it looks like a read-only data, please use STATUS.

Reply:
@venkatmahalingam , just to be aligned, do we need to change the leaf as well? or just the container name?